routers/repo/download.go

```diff
@@ -32,7 +34,11 @@ func ServeData(ctx *context.Context, name string, reader io.Reader) error {
 	// Google Chrome dislike commas in filenames, so let's change it to a space
 	name = strings.Replace(name, ",", " ", -1)
 
-	if base.IsTextFile(buf) || ctx.QueryBool("render") {
+	mimeType := mime.TypeByExtension(filepath.Ext(name))
+
+	if base.IsTextFile(buf) && true == ctx.Data["IsRawApi"] && mimeType != "" {
```

Comment:
Why only text files?

Reply:
Also, what about adding...?:
```
ctx.Resp.Header().Set("Content-Disposition", fmt.Sprintf(`attachment; filename="%s"`, name))
```

However, this should be set only if the file is not a picture or a text file (mime type starts with `text/` or `image/`).

Reply:
I just updated the pull request in order to target all file types. Now, the check for raw API requests is done after the existing Content-Type / Content-Disposition handling.

This way, the Content-Disposition header will be set as requested. In the case of a text file, the Content-Type header might be reset (to a more specific type) by the added code.